cesium/internal/domain/writer_behavior_test.go

```diff
@@ -10,18 +10,39 @@
 package domain_test
 
 import (
-	"github.com/synnaxlabs/cesium/internal/core"
-	"sync"
-
+	"encoding/binary"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	"github.com/samber/lo"
+	"github.com/synnaxlabs/cesium/internal/core"
 	"github.com/synnaxlabs/cesium/internal/domain"
+	xfs "github.com/synnaxlabs/x/io/fs"
 	"github.com/synnaxlabs/x/telem"
 	. "github.com/synnaxlabs/x/testutil"
 	"github.com/synnaxlabs/x/validate"
+	"os"
+	"sync"
+	"time"
 )
 
+func extractPointer(f xfs.File) (p struct {
```

Comment:
This is ok, and I get why you've created an anomymous struct instead of making pointer public, but ngl maybe we should make pointer public

Reply:
I think unless we need to use something private in the actual code, there is no need to making that thing public for the sake of testing – I think a smaller interface is probably better